File: M_Trivia/M_Trivia/GetGameResultsResponse.cpp
#include "GetGameResultsResponse.hpp"

GetGameResultsResponse::GetGameResultsResponse(unsigned int status, const std::vector<PlayerResults>& results) :
	Response(status), _results(results)
{
}

ResponseCodes GetGameResultsResponse::getCode() const
{
	return ResponseCodes::C_GetGameResultsResponse;
}

nlohmann::json GetGameResultsResponse::getJson() const
{
	nlohmann::json j = Response::getJson();
	j["Results"] = _results;
	return j;
}

File: ClientApp/ClientApp/Models/Responses/GetGameResultsResponse.cs
using ClientApp.Models.ResponseErrors;
using Newtonsoft.Json;

namespace ClientApp.Models.Responses
{
    class GetGameResultsResponse : Response<NoResponseErrors>
    {
        private GetGameResultsResponse() { }

        [JsonProperty]
        public List<PlayerResults> Results { get; private set; }

    }
}
